packages/blade/src/components/Carousel/__tests__/Carousel.web.test.tsx

```diff
@@ -0,0 +1,437 @@
+/* eslint-disable @typescript-eslint/no-unnecessary-type-assertion */
+/* eslint-disable @typescript-eslint/require-await */
+import { fireEvent, act } from '@testing-library/react';
+import { Carousel } from '../Carousel';
+import { CarouselItem } from '../CarouselItem';
+import { useCarouselContext } from '../CarouselContext';
+import renderWithTheme from '~utils/testing/renderWithTheme.web';
+
+const TestimonialCard = ({ index }: { index: number }): React.ReactElement => {
+  const { activeSlide } = useCarouselContext();
+  return (
+    <div>
+      <p>{index}</p>
+      {/* A hack to get the internal state of the carousel so that i can assert the tests */}
+      {/* Users won't be able to do this since useCarouselContext is not gonna be exported */}
+      <p data-testid="active-slide">{activeSlide}</p>
+    </div>
+  );
+};
+
+beforeAll(() => {
+  Object.defineProperty(window.Element.prototype, 'scroll', {
+    writable: true,
+    value: jest.fn(),
+  });
+
+  Object.defineProperty(window.Element.prototype, 'scrollLeft', {
+    writable: true,
+    value: jest.fn(),
+  });
+
+  Object.defineProperty(window.Element.prototype, 'getBoundingClientRect', {
+    writable: true,
+    value: jest.fn(() => ({ width: 100, left: 0 })),
+  });
+});
+
+describe('<Carousel />', () => {
+  it('should go to next/previous slide', () => {
+    const onChange = jest.fn();
+    const { getByRole, queryAllByRole, queryAllByTestId } = renderWithTheme(
+      <Carousel visibleItems={1} onChange={onChange}>
+        <CarouselItem>
+          <TestimonialCard index={0} />
+        </CarouselItem>
+        <CarouselItem>
+          <TestimonialCard index={1} />
+        </CarouselItem>
+        <CarouselItem>
+          <TestimonialCard index={1} />
```

Comment:
Why does multiple `TestimonialCard` have the same index?

Reply:
Not really using it for anything, removed